test/test_wavefront.py

```diff
@@ -1,56 +1,61 @@
-import unittest
 import os
+import unittest
 
 import pywavefront
 
+
 def prepend_dir(file):
     return os.path.join(os.path.dirname(__file__), file)
 
+
 class TestWavefront(unittest.TestCase):
     def setUp(self):
         self.mesh_names = ['Simple', 'SimpleB']
         self.material_names = ['Material.simple', 'Material2.simple']
         self.meshes = pywavefront.Wavefront(prepend_dir('simple.obj'))
 
     def testMaterials(self):
-        "Ensure parsed wavefront materials match known values."
+        """Ensure parsed wavefront materials match known values."""
         self.assertEqual(len(self.meshes.materials), len(self.material_names))
         self.assertEqual(self.meshes.materials[self.material_names[0]].__class__,
-                        pywavefront.material.Material)
+                         pywavefront.material.Material)
 
     def testMeshes(self):
-        "Ensure parsed wavefront meshes match known values."
+        """Ensure parsed wavefront meshes match known values."""
         self.assertEqual(len(self.meshes.meshes), len(self.mesh_names))
         self.assertEqual(self.meshes.meshes[self.mesh_names[0]].__class__,
-                pywavefront.mesh.Mesh)
+                         pywavefront.mesh.Mesh)
 
     def testMeshList(self):
-        "Ensure parsed wavefront mesh list matches known values."
+        """Ensure parsed wavefront mesh list matches known values."""
         self.assertEqual(len(self.meshes.mesh_list), len(self.mesh_names))
         self.assertEqual(self.meshes.mesh_list[0].__class__,
                 pywavefront.mesh.Mesh)
 
     def testAddDuplicateMesh(self):
-        "Adding a duplicate mesh should increment the mesh list, but not the meshes hash."
+        """Adding a duplicate mesh should increment the mesh list, but not the meshes hash."""
         self.meshes.add_mesh(self.meshes.meshes[self.mesh_names[0]])
         self.assertEqual(len(self.meshes.meshes), len(self.mesh_names))
         self.assertEqual(len(self.meshes.mesh_list), len(self.mesh_names) + 1)
 
     def testMeshMaterialVertices(self):
-        "Mesh vertices should have known values."
+        "" "Mesh vertices should have known values."""
         self.assertEqual(len(self.meshes.meshes[self.mesh_names[0]].materials[0].vertices), 24)
 
+
 class TestBrokenWavefront(unittest.TestCase):
 
     def testUnknownUsemtl(self):
-        "Referencing an unknown material with usemtl should raise an exception."
+        """Referencing an unknown material with usemtl should raise an exception."""
         self.assertRaises(pywavefront.PywavefrontException,
-                pywavefront.Wavefront, prepend_dir('simple_unknown_usemtl.obj'))
+                          pywavefront.Wavefront,
+                          prepend_dir('simple_unknown_usemtl.obj'))
+
+    # def testMissingNormals(self):
```

Comment:
Is there any documentation or supporting argument on behavior without normals here?

Do we want to at least show a warning? Or is the specification permissive? The original use case was that missing normals caused a black screen on render. Do we ignore that, or somehow handle or warn on it?

Reply:
The plan after this was to make pyglet draw stuff without normals as well. This warning or error check only makes sens to people using the visualization part of this library. We can sort this out before you make a new release.

Reply:
I realized I gave you a shitty answer on this. Here are snippets from the spec:
> Optionally, texture coordinate indices can be used to specify texture coordinates when defining a face.

> Optionally, normal indices can be used to specify normal vectors for vertices when defining a face.

Reply:
No problem; thanks for clarifying!